.github/workflows/release.yml

```
@@ -0,0 +1,81 @@
+name: release
+
+on:
+  workflow_dispatch:
+  push:
+    tags:
+      - '*.*.*'
+
+jobs:
+  release:
+    name: Create a github branch
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      # Create actual release
+
+      - name: Create GitHub release
+        uses: ncipollo/release-action@v1
+        with:
+          bodyFile: notes/${GITHUB_REF/refs\/tags\//}.md"
+          token: ${{ secrets.GH_TOKEN }}
+
+      - name: Get the branch version
+        id: get_branch
+        run: |
+          BRANCH=${GITHUB_REF/refs\/tags\//}
+          if ${{github.event_name == 'workflow_dispatch'}}; then
+            BRANCH=$(git describe --abbrev=0 --tags)
+          fi
+          echo ::set-output name=BRANCH::$(echo ${BRANCH} | cut -d '.' -f1-2 | tr -d 'v')
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+        id: go
+
+      - name: Bump main version
+        run: make patch
+
+      - name: Create a GitHub branch
+        uses: peterjgrainger/action-create-branch@v2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
+        with:
+          branch: ${{ steps.get_branch.outputs.BRANCH }}
+
+      - name: Commit changes to the minor branch
+        uses: EndBug/add-and-commit@v7
+        with:
+          default_author: github_actor
+          message: 'Update patch version'
+          branch: ${{ steps.get_branch.outputs.BRANCH }}
+
+  bump-main-version:
```

```suggestion
  bump-minor-version:
```
I think represents what this is actually doing better, "main" could easily be interpreted as "major"